appcelerator/titanium_mobile
# feat(android): support touch feedback for all backgrounds or no background

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-26315

**Summary:**
- The `Ti.UI.View` properties `touchFeedback` and `touchFeedbackColor` used to only show a tap ripple effect when a `backgroundColor` was applied to the view. This was a documented limitation.
- Now shows touch feedback ripple to the following as well:
  * backgroundImage
  * backgroundGradient
  * backgroundColor set to TRANSPARENT
  * no background
- Note that touch feedback is only applied to a view's background.

**Test:**
1. Build and run the below on Android 5.0 or higher.
2. Tap on "Transparent Background" and verify a yellow ripple appears.
3. Tap on "White Background" and verify a yellow ripple appears.
4. Tap on "Gradient Background" and verify a yellow ripple appears.
5. Tap on "Image Background" and verify a yellow ripple appears only on the image.
6. Tap on "Custom Button" and verify a yellow ripple appears within the blue button.

```javascript
var window = Ti.UI.createWindow();
var scrollView = Ti.UI.createScrollView({
	layout: "vertical",
	scrollType: "vertical",
	showHorizontalScrollIndicator: false,
	shorVerticalScrollIndicator: true,
	width: Ti.UI.FILL,
	height: Ti.UI.FILL,
});
scrollView.add(Ti.UI.createLabel({
	text: "\nTransparent\nBackground\n",
	textAlign: Ti.UI.TEXT_ALIGNMENT_CENTER,
	touchFeedback: true,
	touchFeedbackColor: "yellow",
	top: "10dp",
	width: "75%",
}));
scrollView.add(Ti.UI.createLabel({
	text: "\nWhite\nBackground\n",
	textAlign: Ti.UI.TEXT_ALIGNMENT_CENTER,
	color: "black",
	backgroundColor: "white",
	touchFeedback: true,
	touchFeedbackColor: "yellow",
	top: "10dp",
	width: "75%",
}));
scrollView.add(Ti.UI.createLabel({
	text: "\nGradient\nBackground\n",
	textAlign: Ti.UI.TEXT_ALIGNMENT_CENTER,
	color: "white",
	backgroundGradient: {
		type: "linear",
		startPoint: { x: "0%", y: "50%" },
		endPoint: { x: "100%", y: "50%" },
		colors: [ { color: "white", offset: 0.0}, { color: "blue", offset: 1.0 } ],
	},
	touchFeedback: true,
	touchFeedbackColor: "yellow",
	top: "10dp",
	width: "75%",
}));
scrollView.add(Ti.UI.createLabel({
	text: "\nImage\nBackground\n",
	textAlign: Ti.UI.TEXT_ALIGNMENT_CENTER,
	color: "green",
	backgroundImage: "/appicon.png",
	touchFeedback: true,
	touchFeedbackColor: "yellow",
	top: "10dp",
	width: "100dp",
	height: "100dp",
}));
scrollView.add(Ti.UI.createButton({
	title: "Custom Button",
	backgroundImage: "android.resource://android/drawable/panel_picture_frame_bg_focus_blue",
	touchFeedback: true,
	touchFeedbackColor: "yellow",
	top: "10dp",
}));
window.add(scrollView);
window.open();
```
